irods/test/access_test.py

```diff
@@ -93,34 +93,42 @@ def test_set_inherit_and_test_data_objects (self):
             self.sess.permissions.set(acl_read)
             self.sess.permissions.set(acl_inherit)
 
-            # one new object created since ACL's are applied
+            # create one new object and one new collection *after* ACL's are applied
             new_object_path = test_coll_path + "/my_data_obj"
-            self.sess.data_objects.create( new_object_path )
+            with self.sess.data_objects.open( new_object_path ,'w') as f: f.write(b'some_content')
 
-            #import pdb; pdb.set_trace()
+            new_collection_path = test_coll_path + "/my_colln_obj"
+            new_collection = self.sess.collections.create( new_collection_path )
 
-            collids = [c[Collection.id] for c in 
+            coll_IDs = [c[Collection.id] for c in
                             self.sess.query(Collection.id).filter(Like(Collection.name , deepcoll.path + "%"))]
 
-            Lrods = list(self.sess.query(Collection.name,DataObject.name).filter(
-                                                                          In(DataObject.collection_id, collids )))
+            D_rods = list(self.sess.query(Collection.name,DataObject.name).filter(
+                                                                          In(DataObject.collection_id, coll_IDs )))
 
-            self.assertEqual (len(Lrods), OBJ_PER_LVL*DEPTH+1) # counts the 'older' objects plus one new object
+            self.assertEqual (len(D_rods), OBJ_PER_LVL*DEPTH+1) # counts the 'older' objects plus one new object
 
-            with iRODSSession (port=self.sess.port, zone=self.sess.zone, host=self.sess.host, 
+            with iRODSSession (port=self.sess.port, zone=self.sess.zone, host=self.sess.host,
                                user='bob', password='bpass') as bob:
 
-                L = list(bob.query(Collection.name,DataObject.name).filter(
-                                                                    In(DataObject.collection_id, collids )))
+                D = list(bob.query(Collection.name,DataObject.name).filter(
+                                                                    In(DataObject.collection_id, coll_IDs )))
 
-                # - bob should only see the new data object
+                # - bob should only see the new data object, but none existing before ACLs were applied
```

Comment:
you say ACLs here... are you meaning the magical indexing AVUs?

Reply:
We're testing that when an an ACL permission is granted (accompanied by inheritance) to the top level of the test collection, only new objects are available to the grantee.

Reply:
oh, of course, wrong repo - carry on.